sync-android/src/test/java/com/cloudant/sync/datastore/encryption/EndToEndEncryptionTest.java

```diff
@@ -0,0 +1,205 @@
+
+        if(dataShouldBeEncrypted) {
+            assertThat("SQLite magic bytes found in file that should be encrypted",
+                    sqlCipherMagicBytes, IsNot.not(IsEqual.equalTo(magicBytesBuffer)));
+        } else {
+            assertThat("SQLite magic bytes not found in file that should not be encrypted",
+                    sqlCipherMagicBytes, IsEqual.equalTo(magicBytesBuffer));
+        }
+    }
+
+    @Test
+    public void indexDataEncrypted() throws IOException {
+
+        IndexManager im = new IndexManager(this.datastore);
+        im.ensureIndexed(Arrays.<Object>asList("name", "age"));
+
+        File jsonDatabase = new File(datastore_manager_dir
+                + File.separator + "EndToEndEncryptionTest"
+                + File.separator + "extensions"
+                + File.separator + "com.cloudant.sync.query"
+                + File.separator + "indexes.sqlite");
+
+        InputStream in = new FileInputStream(jsonDatabase);
+        byte[] magicBytesBuffer = new byte[sqlCipherMagicBytes.length];
+        int readLength = in.read(magicBytesBuffer);
+
+        assertEquals("Didn't read full buffer", magicBytesBuffer.length, readLength);
+
+        if(dataShouldBeEncrypted) {
+            assertThat("SQLite magic bytes found in file that should be encrypted",
+                    sqlCipherMagicBytes, IsNot.not(IsEqual.equalTo(magicBytesBuffer)));
+        } else {
+            assertThat("SQLite magic bytes not found in file that should not be encrypted",
+                    sqlCipherMagicBytes, IsEqual.equalTo(magicBytesBuffer));
+        }
+    }
+
+    @Test
+    public void attachmentsDataEncrypted() throws IOException, DocumentException {
+
+        MutableDocumentRevision rev = new MutableDocumentRevision();
+        rev.body = DocumentBodyFactory.create(new HashMap<String, String>());
+
+        File expectedPlainText = TestUtils.loadFixture("fixture/EncryptedAttachmentTest_plainText");
+        assertNotNull(expectedPlainText);
+
+        UnsavedFileAttachment attachment = new UnsavedFileAttachment(
+                expectedPlainText, "text/plain");
+        rev.attachments.put("EncryptedAttachmentTest_plainText", attachment);
+
+        datastore.createDocumentFromRevision(rev);
+
+        File attachmentsFolder = new File(datastore_manager_dir
+                + File.separator + "EndToEndEncryptionTest"
+                + File.separator + "extensions"
+                + File.separator + "com.cloudant.attachments");
+
+        File[] contents = attachmentsFolder.listFiles();
+        assertNotNull("Didn't find expected attachments folder", contents);
+        assertThat("Didn't find expected file in attachments", contents.length, IsEqual.equalTo(1));
+        InputStream in = new FileInputStream(contents[0]);
+
+        if(dataShouldBeEncrypted) {
+
+            byte[] actualContent = new byte[expectedFirstAttachmentByte.length];
+            int readLength = in.read(actualContent);
```

Comment:
Unfortunately it's not possible, the next 16 bytes are the random IV and the next the encrypted data. I could check the blob stored isn't just the plaintext, but I'm not sure that buys us more information.

I'd prefer to stick with the buffer method, just because it's more familiar across the code.